Sfs Review

yashwanth686007 opened the pull request.

lokeshbv commented:
Review

manoharahk reviewed (changes requested):
Please change the commit message. Clearly explain what your patch does.
A Sample format as below
<Line 1> subject 
<Line 2> empty line
<Line 3..N-2> body of the subject
<Line N-1> empty line
<Line N> signed-off
Example : https://github.com/Samsung/TizenRT/commit/7db81c8d3f520ae8c904f8c2e075f13459344336

manoharahk reviewed (commented):
Please change the commit message. Clearly explain what your patch does.
A Sample format as below
<Line 1> subject 
<Line 2> empty line
<Line 3..N-2> body of the subject
<Line N-1> empty line
<Line N> signed-off
Example : https://github.com/Samsung/TizenRT/commit/7db81c8d3f520ae8c904f8c2e075f13459344336

manoharahk commented on external/mbedtls/Make.defs:
```diff
@@ -53,4 +53,7 @@
 ifeq ($(CONFIG_NET_SECURITY_TLS),y)
 CONFIGURED_EXT += mbedtls
 endif
+ifeq ($(CONFIG_TC_SFS),y)
```
What if both CONFIG_NET_SECURITY_TLS and CONFIG_TC_SFS is enabled? It will added two times right.

yashwanth686007 replied:
Tried using "or" command. But still facing build errors.

manoharahk commented on external/mbedtls/Makefile:
```diff
@@ -74,12 +77,12 @@ SRC_CRYPTO_CSRCS =    aes.c           aesni.c         arc4.c          \
                       version_features.c              xtea.c
-
+endif
 SRC_X509_CSRCS =      certs.c         pkcs11.c        x509.c          \
```
Do we require SRC_X509_CSRCS  for CONFIG_TC_SFS?

yashwanth686007 replied:
No. Not required. Only aes.c need to be compiled.

manoharahk commented on os/fs/smartfs/smartfs_smart.c:
```diff
@@ -567,10 +577,45 @@ static ssize_t smartfs_read(FAR struct file *filep, char *buffer, size_t buflen)
 
-	/* Return the number of bytes we read */
-
+	final_output = (char *)malloc(buflen + 1);
```
I think it is better to add a new module for ex: smartf_sfs.c which will all the implementation and only very few calls and minor changes will be done to smartfs_smart.c.. 
It may not be good to modify too much inside the existing smartfs_smart implementation.

yashwanth686007 committed:
Add Crypto Library

Adds Crypto library for AES algorithm and to test it's functionality.

This patch is Back Ported from Latest "Nuttx" Source

Signed-off-by: Yashwanth <v.yashwanth@samsung.com>

yashwanth686007 committed:
Secure File System Feature

This patch supports data encryption using AES algorithm in smartfs layer.

Signed-off-by: Yashwanth <v.yashwanth@samsung.com>

yashwanth686007 committed:
Secure File System Testcases

This patch contains testcases related to secure file system with different API's such as fwrite, fread, fgets etc.

Signed-off-by: Yashwanth <v.yashwanth@samsung.com>
